### PR TITLE
feat: add LocalConfigRuntimeState

### LINKED ISSUE #1881: integrate the business layer local config with the data layer local config for reading/writing to the the file system
integrate the business layer local config with the data layer local config for reading/writing to the the file system.  This will also enable the migration logic required to upgrade from v0.35.1+ to v0.36.0.

We want all of the direct integration with the data layer to go through the business layer's [runtime state](https://github.com/hiero-ledger/solo/blob/main/docs/design/architecture/system/architecture_overview.md).  Thus, we will want to create a `src/business/runtime-state/local-config-runtime-state.ts` `LocalConfigRuntimeState` that is injectable to handle the data layer calls.  The existing business remote config will leverage the `LocalConfigRuntimeState` to do the reading/writing via the data layer.  Much if not all of the existing can be restructured and rolled into this runtime state class.  However, we do not need to do all of this at once, as we want to get v0.36.0 published soon and need this integration complete.

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

- This PR adds a `LocalConfigRuntimeState` class, replacing the current `LocalConfig` implementation, and incorporates it throughout solo.
- Updates all `ClusterReferences` to the new Map<string, string> type. 
- The new class will automatically migrate the user's `local-config.yaml` files to the updated structure.
- Removes the UserEmailAddress type and flags.
- Adds a new `username` flag. It is used in `solo init` and provides the user with an option to set their `UserIdentity.name` when creating the `local-config.yaml file` .

### Related Issues

* Closes #1881 

### Pull request (PR) checklist

* \[x] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[x] This PR added no TODOs or commented out code
* \[x] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[x] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[x] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[x] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/commands/node/tasks.ts [hunk 1]
```diff
@@ -2051,11 +2050,16 @@ export class NodeCommandTasks {
...
 
+        const clusterReverencesList: ClusterReference[] = [];
+        for (const [clusterReference] of clusterReferences) {
+          clusterReverencesList.push(clusterReference);
+        }
+
         // Update all charts
         await Promise.all(
-          Object.keys(clusterReferences).map(async clusterReference => {
+          clusterReverencesList.map(async clusterReference => {
```
Comment (instamenta) [inline src/commands/node/tasks.ts:2060]: Typo

File: test/e2e/commands/dual-cluster-full.test.ts [hunk 2]
```diff
@@ -80,6 +80,7 @@ describe('Dual Cluster Full E2E Test', async function dualClusterFullEndToEndTes
   // TODO the kube config context causes issues if it isn't one of the selected clusters we are deploying to
   before(async (): Promise<void> => {
     fs.rmSync(testCacheDirectory, {recursive: true, force: true});
+    fs.mkdirSync(testCacheDirectory, {recursive: true});
```
Comment (jeromy-cannon) [inline test/e2e/commands/dual-cluster-full.test.ts:83]: why are we adding this?  We want to ensure that Solo can create its own dependencies.  If we add the directory here, then we are no longer testing that Solo can create the directories on its own.

File: test/e2e/commands/dual-cluster-full.test.ts [hunk 3]
```diff
@@ -97,6 +98,10 @@ describe('Dual Cluster Full E2E Test', async function dualClusterFullEndToEndTes
   beforeEach(async (): Promise<void> => {
     testLogger.info(`${testName}: resetting containers for each test`);
     resetForTest(namespace.name, testCacheDirectory, testLogger, false);
+    const localConfig: LocalConfigRuntimeState = container.resolve<LocalConfigRuntimeState>(
+      InjectTokens.LocalConfigRuntimeState,
+    );
+    await localConfig.load();
```
Comment (jeromy-cannon) [inline test/e2e/commands/dual-cluster-full.test.ts:104]: I believe this can be removed.  Since we call Solo commands via the main command, then the middleware local config call should be loading this, making it redundant.

File: test/test-utility.ts [hunk 4]
```diff
@@ -134,6 +134,8 @@ interface Cmd {
   deploymentCmdArg?: DeploymentCommand;
 }
 
+let resetDone: boolean = false;
```
Comment (jeromy-cannon) [inline test/test-utility.ts:137]:
```suggestion
let shouldReset: boolean = true;
```

File: src/business/runtime-state/local-config-runtime-state.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+  public getShard(deploymentName: DeploymentName): Shard {
+    return this.getDeployment(deploymentName).shard;
+  }
+
+  public async modify(callback: (modelData: LocalConfig) => Promise<void>): Promise<void> {
+    if (!this.isLoaded()) {
+      throw new WriteLocalConfigBeforeLoadError('Attempting to modify local config before loading it');
+    }
+    await callback(this.source.modelData);
```
Comment (jeromy-cannon) [inline src/business/runtime-state/local-config-runtime-state.ts:103]: we should wrap this in a try catch, especially since, we have no idea what will be executed

**Review by jeromy-cannon — CHANGES_REQUESTED** [review]

**Review by jeromy-cannon — APPROVED** [review]
